sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -646,11 +650,87 @@ internal virtual Task ClearAsync(CancellationToken cancellationToken = default)
         ///   This method will automatically unregister the <see cref="SendEventBatchSucceededAsync"/> and <see cref="SendEventBatchFailedAsync"/> handlers.
         /// </remarks>
         ///
-        public virtual Task CloseAsync(bool flush = true,
-                                       CancellationToken cancellationToken = default)
+        public virtual async Task CloseAsync(bool flush = true,
+                                             CancellationToken cancellationToken = default)
         {
-            _isClosed = true;
-            throw new NotImplementedException();
+            if (_isClosed)
+            {
+                return;
```

Comment:
Do we have a test that exercises this path?

Reply:
Not yet; we should.   I'm still tweaking some of the internals to flesh things out a bit more.   I'll add it before closing this out.

Reply:
Added